### DISCUSSION TITLE
Azure OpenAI support

### DISCUSSION THREAD

**Discussion by pelikhan** [algorithmicsuperintelligence/optillm#11] [General]
Would it be possible to plug an Azure OpenAI client without being too intrusive?

**Comment by codelion** [comment]
At the moment we pass the OpenAI `client` from the proxy to the individual approaches that in turn call the base model. To support a models (deployments) on Azure, we will need to switch to the AzureOpenAI using something like this:

```
client = AzureOpenAI(
  azure_endpoint = os.getenv("AZURE_OPENAI_ENDPOINT"), 
  api_key=os.getenv("AZURE_OPENAI_API_KEY"),  
  api_version="2024-02-01"
)
```
I have opened a ticket (#13) to implement it. Happy to accept PRs as well.

**Comment by codelion** [comment]
This was implemented in #13

**Reply by pelikhan** [reply]
and https://github.com/codelion/optillm/pull/21
